docs/MIGRATION-GUIDE-GENERIC-EXECUTOR-SERVICE.md

```diff
@@ -0,0 +1,241 @@
+* Move the script into another folder (optional)
+
+For more information regarding environment variables, please consult [FEATUERS.md](../FEATURES.md).
+
+**job/config.yaml**
+```yaml
+apiVersion: v2
+actions:
+  - name: "Deploy using bash scripts"
+    events:
+      - name: "sh.keptn.event.deployment.triggered"
+    tasks:
+      - name: "Run bash script"
+        files:
+          - /generic-executor-service/deployment.triggered.sh
+        env:
+          - name: DATA_CONFIGURATIONCHANGE_VALUES_IMAGE
+            value: "$.data.configurationChange.values.image"
+            valueFrom: event
+          - name: DATA_DEPLOYMENT_DEPLOYMENTSTRATEGY
+            value: "$.data.deployment.deploymentstrategy"
+            valueFrom: event
+          - name: my-super-secret
+            valueFrom: secret
+        workingDir: "/keptn"
+        image: "debian:bookworm-slim"
+        cmd: ["bash"]
+        args: ["./generic-executor-service/deployment.triggered.sh"]
+```
+
+*Note*: We are using `debian:bookworm-slim` as an image here. This is a lightweight debian based image, which does not
+provide a lot of tools (it's fine for scripting purpose, e.g., scripting using `echo`, `cat`, `grep` etc..., but not 
+for using tools like `curl`, `wget`, etc... which you would have to install).
+
+You can verify this locally by running
+```console
+docker run -it --rm debian:bookworm-slim bash
+```
+and pasting your script (or parts of it) into the terminal.
+
+
+### Troubleshooting
+
+#### Permission Denied
+
+If you get a message like this:
+```
+bash: line 1: ./generic-executor-service/deployment.triggered.sh: Permission denied
+```
+you need to make sure that your script has the executable permission in the git upstream repo, e.g.: 
+```console
+chmod +x generic-executor-service/deployment.triggered.sh
+keptn add-resource --project=<PROJECT> --service=<SERVICE> --stage=<STAGE> --resource=generic-executor-service/deployment.triggered.sh --resourceUri=generic-executor-service/deployment.triggered.sh
+```
+
+
+## Migrating Python Scripts
+
+In generic-executor-service you can use bash scripts by creating a `$event.sh` file (where `$event` reflects the Cloud
+event type you are listening for), with the following content:
+
+**deployment.triggered.py**
+```python
+import os
+import sys
+
+# Lets get the first parameter which could potentially be a local file name
+methodArg = ""
+if len(sys.argv) > 1:
+    methodArg = sys.argv[1]
+
+print("This is my generic handler script and I got passed " + methodArg + " as parameter")
+print("I also have some env variables, e.g: SHKEPTNCONTEXT=" + os.getenv('SHKEPTNCONTEXT', ""))
+print("SOURCE=" + os.getenv('SOURCE',""))
+print("PROJECT=" + os.getenv('DATA_PROJECT',""))
+```
+
+In general, this can be achieved by re-using existing `.sh` files in job-executor-service, using a container that provides bash and the necessary tools, and providing all necessary variables within the code.
```

Comment:
```suggestion
In general, this can be achieved by re-using existing `.py` files in job-executor-service, using a container that provides python and the necessary tools, and providing all necessary variables within the code.
```